src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Transmission/OtlpExporterPersistentStorageTransmissionHandler.cs

```diff
@@ -0,0 +1,175 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+using Google.Protobuf;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+using OpenTelemetry.PersistentStorage.Abstractions;
+using OpenTelemetry.PersistentStorage.FileSystem;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Transmission;
+
+internal sealed class OtlpExporterPersistentStorageTransmissionHandler<TRequest> : OtlpExporterTransmissionHandler<TRequest>, IDisposable
+{
+    internal int RetryIntervalInMilliseconds = 60000;
+    private readonly ManualResetEvent shutdownEvent = new(false);
+    private readonly ManualResetEvent dataExportNotification = new(false);
+    private readonly AutoResetEvent exportEvent = new(false);
+    private readonly Thread thread;
+    private readonly PersistentBlobProvider persistentBlobProvider;
+    private readonly Func<byte[], TRequest>? requestFactory;
+    private bool disposed;
+
+    public OtlpExporterPersistentStorageTransmissionHandler(IExportClient<TRequest> exportClient, double timeoutMilliseconds, Func<byte[], TRequest> requestFactory, string storagePath)
+        : this(new FileBlobProvider(storagePath), exportClient, timeoutMilliseconds, requestFactory)
+    {
+    }
+
+    internal OtlpExporterPersistentStorageTransmissionHandler(PersistentBlobProvider persistentBlobProvider, IExportClient<TRequest> exportClient, double timeoutMilliseconds, Func<byte[], TRequest> requestFactory)
+        : base(exportClient, timeoutMilliseconds)
+    {
+        this.persistentBlobProvider = persistentBlobProvider;
+        this.requestFactory = requestFactory;
+
+        this.thread = new Thread(this.RetryStoredRequests)
+        {
+            Name = $"OtlpExporter Persistent Retry Storage - {typeof(TRequest)}",
+            IsBackground = true,
+        };
+
+        this.thread.Start();
+    }
+
+    // Used for test.
+    internal bool Forceflush(int timeOutMilliseconds)
+    {
+        this.exportEvent.Set();
+
+        if (this.dataExportNotification.WaitOne(timeOutMilliseconds))
```

Comment:
Seems like this will cause a retry of up to 10 files not necessarily a flush of everything. If it is a test thing maybe just rename it for clarity? `InitiateAndWaitForRetryProcess` 🤷